### DISCUSSION TITLE
Github.com rate limit for tofu-* actions

### DISCUSSION THREAD

**Discussion by cberge908** [dflook/terraform-github-actions#331] [Q&A] [answered]
Hi all,

we're running our environment on GHES (not github.com). We've recently switched over to use OpenTofu instead of Terraform. 
The actions are working perfectly fine, but from time to time we're hitting rate limits on github.com while the action is fetching the opentofu binary.

Is there any option to supply an additional token to overcome these rate limits?

Cheers,
@cberge908

**Comment by dflook** [comment]
Is there anything in the workflow log that indicates where the rate limit occurs?

**Reply by cberge908** [reply]
Yep, directly after installing:

```log
Installing OpenTofu
  API rate limit exceeded for x.x.x.x. (But here's the good news: Authenticated requests get a higher rate limit. Check out the documentation for more details.) Try again when the rate limit resets at 24-02-28 15:51:26 UTC.
```

**Comment by dflook** [comment, marked as answer]
v1.42.0 has just been released which has a new `GITHUB_DOT_COM_TOKEN` environment variable, which you can set to hopefully avoid the rate limit.

I'm not able to test it, so please let me know if it works.

**Reply by cberge908** [reply]
Awesome - thanks for the quick reaction. Just saw the release notes. Will give this a try early next week and report if this worked properly :)

**Reply by cberge908** [reply]
So far we had the action run a few times today, no rate limit issues so far. Will keep an eye on that for the rest of the week. If there is no further occurrence of this we can mark this as closed.

**Reply by cberge908** [reply]
We've tested it our trough the whole week, not a single error due to API rate limits yet.
I guess we can close this one - thanks @dflook once again for the quick help here :)
